wp-decoupled-preview.php

```diff
@@ -61,7 +61,7 @@ function conditionally_enqueue_scripts() {
 
 	if ( 'post.php' === $pagenow || 'post-new.php' === $pagenow ) {
 		add_action( 'admin_bar_menu', __NAMESPACE__ . '\\add_admin_decoupled_preview_link', 100 );
-		add_action( 'admin_enqueue_scripts', __NAMESPACE__ . '\\enqueue_scripts' );
+		add_action( 'enqueue_block_editor_assets', __NAMESPACE__ . '\\enqueue_scripts', 100 );
```

Comment:
Are there cases where a user would not be using Gutenberg? I think if they are using the Classic Editor this will not load the js (since `enqueue_block_editor_assets` only runs on G'berg posts) but I also don't know the degree to which non-gutenberg posts/post types have been tested/are supported.

Reply:
Nice catch. The bug that we're trying to address here is a case where our JS fails because the preview button doesn't yet exist. The default preview button seems to be added (or at least manipulated) by js as well. This change seems to ensure that our script would load well after the preview button exists. But you're right, this does now depend on the block editor. My worry is that if we enqueue with the admin script but change the weight we'll still be in race condition territory (I'm only able to duplicate this on certain machines.)

There is no reason that this plugin wouldn't work with the classic editor, but we haven't been testing it. So I'm inclined to list that as a known issue (I can make that update when queuing up the next dev release) and see if we get any feedback.

Thoughts?

Reply:
I have not tested, but there is a `is_block_editor` conditional that it seems like could work here. Instead of race-conditioning, you load via `admin_enqueue_scripts` if `! is_block_editor()` and via `enqueue_block_editor_assets` if so.

http://developer.wordpress.org/reference/classes/wp_screen/is_block_editor/

Reply:
Something like this maybe?

```suggestion
		$enqueue_hook = is_block_editor() ? 'enqueue_block_editor_assets' : 'admin_enqueue_scripts';
		add_action( $enqueue_hook, __NAMESPACE__ . '\\enqueue_scripts', 100 );
```